hw/ip/otbn/dv/otbnsim/sim/state.py

```diff
@@ -258,20 +293,6 @@ def peek_call_stack(self) -> List[int]:
         '''Return the current call stack, bottom-first'''
         return self.gprs.peek_call_stack()
 
-    def stop(self, err_code: Optional[int]) -> None:
-        '''Set flags to stop the processor.
-
-        If err_code is not None, it is the value to write to the ERR_BITS
-        register.
-
-        '''
-        # INTR_STATE is the interrupt state register. Bit 0 (which is being
-        # set) is the 'done' flag.
-        self.ext_regs.set_bits('INTR_STATE', 1 << 0)
-        # STATUS is a status register. Bit 0 (being cleared) is the 'busy' flag
-        self.ext_regs.clear_bits('STATUS', 1 << 0)
-
-        if err_code is not None:
-            self.ext_regs.write('ERR_BITS', err_code, True)
-
-        self.running = False
+    def on_error(self, error: Alert) -> None:
+        '''Add a pending error that we have seen an illegal instruction'''
```

Comment:
I think is called in two places, one for an illegal instruction, the other for a zero loop count. Maybe it's needs a slightly more generic doc string (perhaps unknown encoding and zero loop count both count as illegal instructions but as illegal instruction is a distinct error from the loop error you get with a zero loop count it could be confusing to use the term 'illegal instruction' here).

Reply:
Ah, this is just stale from a previous version of the code (which generated the `IllegalInsnError` there). Will fix, thanks.